platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygComposerViewModel.swift

```diff
@@ -309,28 +321,57 @@ public extension WysiwygComposerViewModel {
             return true
         }
         
+        let nextTextUpdate = ReplaceTextUpdate(date: Date.now, range: range, text: replacementText)
+        // This is to specifically to work around an issue when tapping on an inline predictive text suggestion within the text view.
+        // Even though we have the delegate disabled during modifications to the textview we still get some duplicate
+        // calls to this method in this case specifically. It's very unlikely we would get a valid subsequent call
+        // with the same range and replacement text within such a short period of time, so should be safe.
+        if let lastReplaceTextUpdate, lastReplaceTextUpdate.isDuplicate(of: nextTextUpdate) {
+            return true
+        }
+        
         // This fixes a bug where the attributed string keeps link and inline code formatting
         // when they are the last formatting to be deleted
         if textView.attributedText.length == 0 {
             textView.typingAttributes = defaultTextAttributes
         }
 
         let update: ComposerUpdate
-        let shouldAcceptChange: Bool
+        let skipTextViewUpdate: Bool
 
+        // The system handles certain auto-compelete use-cases with somewhat unusual replacementText/range
+        // combinations, some of those edge cases are handled below.
+        
```

Comment:
The `let isReplacingWordWithSuggestion` and `if isReplacingWordWithSuggestion` could go up here before the other 2 properties to help follow the logic when reading through.